buildlib/tools/test_malloc_hook.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash -eExl
+realdir=$(realpath $(dirname $0))
+source ${realdir}/../az-helpers.sh
+
+#
+# Prepare build environment
+#
+WORKSPACE=${WORKSPACE:=$PWD}
+ucx_inst=${WORKSPACE}/install
+
+prepare() {
+	echo " ==== Prepare ===="
+	env
+	cd ${WORKSPACE}
+	if [ -d build-test ]
+	then
+		chmod u+rwx build-test -R
+		rm -rf build-test
+	fi
+	mkdir -p build-test
+	cd build-test
+}
+
+#
+# try load cuda modules if nvidia driver is installed
+#
+try_load_cuda_env() {
+	have_cuda=no
+	if [ -f "/proc/driver/nvidia/version" ]; then
+		have_cuda=yes
+	fi
+	if [ "X$have_cuda" == "Xno" ]
+	then
+		azure_log_error "Cuda not found"
+		exit 1
+	fi
+}
```

Comment:
1. rename the function to check_cuda_driver
2. can simplify: if "/proc/driver/nvidia/version" not found, exit with error